src/mail-app/mail/model/MailModel.ts

```diff
@@ -445,8 +446,15 @@ export class MailModel {
 		)
 	}
 
+	async loadAndApplyLabels(mails: () => Promise<readonly Mail[]>, addedLabels: readonly MailFolder[], removedLabels: readonly MailFolder[]) {
```

Comment:
I mentioned before that I think this method does not fit into model because we already have `applyLabels()`. Maybe it could live somewhere in `mailGuiUtils`?